squiRL/vpg/vpg.py

```diff
@@ -148,13 +153,23 @@ def training_step(self, batch: Tuple[torch.Tensor, torch.Tensor],
             replay data
             nb_batch (TYPE): Current index of mini batch of replay data
         """
-        states, actions, rewards, dones, _ = batch
-        states = torch.cat(states)
-        ind = [len(s) for s in dones]
+        states, actions, rewards, firsts, _ = batch
+        ind = torch.nonzero(firsts).squeeze().numpy().tolist() + [
+            firsts.shape[0]
+        ]
+        ind = np.add(ind, 1).tolist()
+        ind = [ind[0]] + [i - j for j, i in zip(ind, ind[1:])]
+        ind = ind[1:]
+
+        actions = torch.split(actions, ind)
+        rewards = torch.split(rewards, ind)
+        firsts = torch.split(firsts, ind)
         action_logits = torch.split(self.net(states.float()), ind)
         episode_rewards = []
         loss = 0
-        for ep in range(self.hparams.episodes_per_batch):
+        for ep in range(len(actions)):
+            if rewards[ep].shape[0] == 1:
+                continue
             episode_rewards.append(rewards[ep].sum().item())
             loss += self.vpg_loss(
```

Comment:
vpg_loss arguments must be updated because right now it's expecting Tuple[torch.Tensor,Tensor])

Reply:
Done. `Tuple([torch.Tensor, torch.Tensor, torch.Tensor])`